client/my-sites/stats/hooks/use-notice-visibility-query.ts

```diff
@@ -65,24 +66,33 @@ const queryNotices = async function ( siteId: number | null ): Promise< Notices
 
 const useNoticesVisibilityQueryRaw = function < T >(
 	siteId: number | null,
-	select?: ( payload: Notices ) => T
+	select?: ( payload: Notices ) => T,
+	staleTime?: number
 ) {
 	return useQuery( {
 		queryKey: [ 'stats', 'notices-visibility', 'raw', siteId ],
 		queryFn: () => queryNotices( siteId ),
 		select,
-		staleTime: 1000 * 30, // 30 seconds
+		staleTime: staleTime !== undefined ? staleTime : 1000 * 30, // custom or 30 seconds
```

Comment:
The method works! However to use `staleTime` as a parameter will override how the query works in all places that use the hook, which I don't think is ideal.

> The query will automatically update when this key changes (as long as enabled is not set to false).

In react-query, `useQuery` has a `queryKey` and `useMutation` has a `mutationKey`, when the two keys match, the mutation will cause the query to be forcibly refreshed. More info: https://tanstack.com/query/latest/docs/framework/react/reference/useQuery

An alternative approach could be that we should be able to get `refetch` handle from `useQuery` and refresh the data proactively after a mutation action:

`const { refetch } = useNoticeVisibilityQuery(...);`

Reply:
Thanks for looking at this @kangzj. I don't think the proposed change would affect other queries because custom time was added as a parameter. Existing code using the query doesn't provide custom time and would use the old value.

Reply:
I could be wrong but from my understanding the queries are shared between different components, so if we override staleTime from one place then all the other places would be affected. And another thing is, we cannot assure which place would be run last; in this particular case, it works, but it might not in other cases or in the future if more components are using the hook. 🤔 

Might be related:

https://github.com/TanStack/query/blob/62704ce63a2c88c8005e037825060f33404cf74b/packages/query-core/src/queryObserver.ts#L194

https://github.com/TanStack/query/blob/62704ce63a2c88c8005e037825060f33404cf74b/packages/query-core/src/queryClient.ts#L320

https://github.com/TanStack/query/blob/62704ce63a2c88c8005e037825060f33404cf74b/packages/query-core/src/queryClient.ts#L122